src/scripts/settings.js

```diff
@@ -0,0 +1,167 @@
+const selectors = {
+    stateJson: '#state',
+    emptyState: '.empty-state',
+    deleteFormula: 'trash',
+    createFormula: 'create_formula',
+    pen: 'pen',
+    textInputforCreatingFormula: '.text_input_lol',
+    discard: '.discard',
+    save: '.save',
+    formulasFolder: '#formulasFolder',
+    template: '.formula-template',
+    createFormulaFromEmptyState: 'b_0',
+    link: 'linkToFormulasPage',
+    table: '.formulas_s',
+    home: '#home'
+}
+
+
+function getState() {
+    const data = localStorage.getItem('state');
+    return JSON.parse(data);
+}
+function updateState(state) {
+    localStorage.setItem('state', JSON.stringify(state));
+}
+
+const state = getState();
+if (state){
+    console.log("🔥#GOOD#🔥 ---> state.json parsed from localeStorage!!!")
+}
+
+
+
+
+
+
+
+//  CURRENT LIST OF FORMULAS
+const currentSettings  = document.querySelector(selectors.formulasFolder);
+//  SAVED LIST OF FORMULAS
+const savedSettings = currentSettings.cloneNode(currentSettings);
+
+const empty = document.querySelector(selectors.emptyState);
+const table = document.querySelector(selectors.table)
+
+// CURRENT URL
+const url = new URL(window.location);
+
+
+// INITIALIZING THE EMTY STATE
+
+
+
+
+//  ADDING EMPTY STATE HIDDEN IF LIST OF FORMULAS IS NOT EMPTY
+
+function emptyStateFunction( x ){
```

Comment:
This function is not needed. This feature could be implemented inside the component itself. But who knows what is a better approach